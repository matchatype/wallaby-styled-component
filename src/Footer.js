import React from "react";
import styled from "styled-components";

const StyledFooter = styled.footer`
  display: flex;
  flex-direction: column;
  max-width: ${({ theme }) =>
    console.log(theme.breakpoints.lg) || theme.breakpoints.lg};
  margin-left: auto;
  margin-right: auto;
  padding: ${({ theme }) => theme.fontSizes.small};
`;

const Footer = ({ siteTitle }) => (
  <StyledFooter>
    {siteTitle} &ndash; Â© {new Date().getFullYear()}. All Rights Reserved.
  </StyledFooter>
);

export default Footer;
