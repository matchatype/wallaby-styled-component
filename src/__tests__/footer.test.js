import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import Footer from "../footer";

describe("Footer", () => {
  let renderer;

  beforeEach(() => {
    renderer = createRenderer();
  });

  it("should render correctly", () => {
    renderer.render(<Footer siteTitle="Test" />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it("should render default props correctly", () => {
    renderer.render(<Footer />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
