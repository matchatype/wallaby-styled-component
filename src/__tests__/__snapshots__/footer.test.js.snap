// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<ForwardRef>
  Test
   – © 
  2019
  . All Rights Reserved.
</ForwardRef>
`;

exports[`Footer should render default props correctly 1`] = `
<ForwardRef>
   – © 
  2019
  . All Rights Reserved.
</ForwardRef>
`;
